### PR TITLE
Use generated config.hh and deprecated.hh rather than the custom ones.

### PR DESCRIPTION
**Pull request by francois-keith**

### CODE REVIEW

**Review by jcarpent — APPROVED** [review]
@francois-keith Good to have you on board. Small question: for which purpose do you use TSID?

File: include/tsid/tasks/task-joint-posture.hpp [hunk 1]
```diff
@@ -21,7 +21,7 @@
 #include <tsid/tasks/task-motion.hpp>
 #include <tsid/trajectories/trajectory-base.hpp>
 #include <tsid/math/constraint-equality.hpp>
-#include <tsid/deprecation.hpp>
+#include <tsid/deprecated.hh>
```
Comment (andreadelprete) [inline include/tsid/tasks/task-joint-posture.hpp:24]: Why this change of name? Is it just a convention or is there some good reason?

Comment (jcarpent) [reply include/tsid/tasks/task-joint-posture.hpp:24]: These files are automatically generated by the cmake module.
The defaults extensions is .hh. You can personalize them using a specific macro.

Comment (francois-keith) [reply include/tsid/tasks/task-joint-posture.hpp:24]: I think this is the convention. The file deprecated.hh is the one generated by the cmake submodule (as well as config.hh), this will make the updates on those file easier to backport.

Comment (andreadelprete) [reply include/tsid/tasks/task-joint-posture.hpp:24]: Got it, thanks for the explanation!
